.github/workflows/ci-app-infra-service.yml

```diff
@@ -16,11 +16,26 @@ on:
       - infra/test/**
       - .github/workflows/ci-app-infra-service.yml
   workflow_dispatch:
+    inputs:
+      version:
+        required: true
+        default: "main"
+        description: Tag or branch or SHA to test
 
 jobs:
+  # `make infra-test-service` will build and publish the release itself, but
+  #  doing it explicitly with the re-usable workflow helps avoid race conditions
```

Comment:
Actually thinking about this more, seems like with this change we should remove the build and publish code from infra-test-service and have that infra-test-service rely on the fact that it's already built. Infra test service is meant to just test the terraform apply and terraform destroy anyways. The build and publish can be done outside of that test suite.

Benefit of removing is to avoid making this workflow take longer by having to build twice